gax/src/test/java/com/google/api/gax/retrying/DirectRetryingExecutorTest.java

```diff
@@ -48,4 +52,32 @@
         new TestResultRetryAlgorithm<String>(apocalypseCountDown, apocalypseException),
         new ExponentialRetryAlgorithm(retrySettings, CurrentMillisClock.getDefaultClock()));
   }
+
+  @Test
+  public void testFutureContainsRetryContext() {
+    RetrySettings retrySettings =
+        RetrySettings.newBuilder()
+            .setInitialRetryDelay(Duration.ofMillis(10))
+            .setRetryDelayMultiplier(1.5)
+            .setMaxRetryDelay(Duration.ofSeconds(10))
+            .setInitialRpcTimeout(Duration.ofMillis(10))
+            .setMaxRpcTimeout(Duration.ofMillis(10))
+            .build();
+    RetryAlgorithm<String> retryAlgorithm = getAlgorithm(retrySettings, 0, null);
+    RetryingExecutorWithContext<String> executor = getExecutor(retryAlgorithm);
+
+    Callable<String> noopCallable =
+        new Callable<String>() {
+          @Override
+          public String call() {
+            return null;
+          }
+        };
+
+    RetryingContext ctx = RetryingContext.createDefault();
+    BasicRetryingFuture<String> future =
+        (BasicRetryingFuture<String>) executor.createFuture(noopCallable, ctx);
+
+    Truth.assertThat(future.getRetryingContext()).isSameAs(ctx);
```

Comment:
This does not look like a useful test.

Reply:
It makes sure that the context makes it to the future which needs it. Otherwise there is nothing preventing someone deleting that code

Reply:
It makes sure that the context makes it to the future which needs it. Otherwise there is nothing preventing someone deleting that code

Reply:
Yes, but is it what we want to test? The fact that it gets to the the future should be tested by the usage of the context (if that will be in subsequent commit, when actual opencensus thing is there, then fine). If the context is not used, the fact that it exist is not important (passed or not passed makes not impact on the library correctness/usability, thus testing it is not important). Here we are trying to test implementation detail and only implementation detail. Please strongly considder just removing the test (also automatically addresses the other test-related comments). 

The retrying context will be tested by the tests, which actually test that the context-specific logic.